solvebio/query.py

```diff
@@ -264,7 +264,8 @@ def __len__(self):
                  SELECT * FROM <table> [WHERE condition] [LIMIT number]
               )
         """
-        return min(self._limit, self.count()) if not self.count() is None else self._limit
+        _limit = 10 ** 15 if self._limit == float('inf') else self._limit
```

Comment:
I guess the issue here is that because we don't have `total`, it tries to send `inf` to the API and fails somewhere? So we need to set a large value for limit here. Are there any other options instead of setting to 10**15?

Reply:
What happens if we use  `None`?

Reply:
for example `__len__` is called here: https://github.com/solvebio/solvebio-python/blob/42e8aab4334c9bc71a88744b7f67db8bc3ea198e/solvebio/query.py#L413

and `None` raises an error `TypeError: 'NoneType' object cannot be interpreted as an integer`.

Honestly, I do not have an idea of what we can add there instead of some large value?

Reply:
What if we handle both cases where total and _limit are both None in the iterator, instead of wrapping it up in `len(self)`?

Reply:
It means that we have to do the same wherever `len(self)` is currently called in code, right?

Reply:
What if in __len__ we `return None if _limit is None and self.count() is None`

Reply:
it is possible, but as far as I know instead of calling `len(self)` we have to call it like `self.__len__()` otherwise it will raise `TypeError: 'NoneType' object cannot be interpreted as an integer`

Reply:
Ok can you set this magic value as a constant in the function `INF = 10**15` and add a comment above it to explain

Reply:
Actually, last thing. Can we just set the initial value of limit to INF instead of float(inf)?